src/pages/Admin/Charity/Permissions/Table.tsx

```diff
@@ -0,0 +1,141 @@
+import { Controller, useFormContext } from "react-hook-form";
+import { useAdminResources } from "pages/Admin/Guard";
+import { useGetWallet } from "contexts/WalletContext";
+import TableSection, { Cells } from "components/TableSection";
+import { FormField, FormValues } from "./schema";
+
+export default function Table({ className = "", disabled = false }) {
+  const { endow_type } = useAdminResources<"charity">();
+  const { wallet } = useGetWallet();
+
+  const { control, getValues } = useFormContext<FormValues>();
+
+  const { initialValues, ...formValues } = getValues();
+
+  const isNormal = endow_type === "normal";
+
+  return (
+    <table
+      className={`${className} w-full text-sm rounded border border-separate border-spacing-0 border-prim`}
+    >
+      <TableSection
+        type="thead"
+        rowClass="bg-orange-l6 dark:bg-blue-d7 divide-x divide-prim"
+      >
+        <Cells
+          type="th"
+          cellClass="py-3 px-4 text-xs uppercase font-semibold text-left first:rounded-tl last:rounded-tr"
+        >
+          <>Action</>
+          <>Admin wallet</>
+          {isNormal ? <>Governance</> : null}
+          <>Delegate</>
+          <>Delegate address</>
+          {isNormal ? <>Actions</> : null}
+        </Cells>
+      </TableSection>
+      <TableSection
+        type="tbody"
+        rowClass="dark:bg-blue-d6 divide-x divide-prim"
+        selectedClass="bg-orange-l5 dark:bg-blue-d4"
+      >
+        {(
+          Object.keys(formValues) as (keyof Omit<FormValues, "initialValues">)[]
+        ).map((formField) => (
+          <Controller
```

Comment:
nit: instead of `<Controller/>` (controlled), and manually setting `onChange` on each `<input/>` could abstract a row
```tsx
//Row.tsx
const { register } = useFormContext()
return <>
   ....
   <input {...register("name.delegate")}/> 
   <input {...register("name.address")}/> 
  </>
```